core/src/buffer.rs

```diff
@@ -88,8 +91,8 @@ impl Buffer {
         if start <= s.len() && end <= s.len() && start <= end {
             Some(Slice {
                 buf: Self(self.0.clone()),
-                start: start,
-                end: end,
+                start,
+                end,
```

Comment:
This was probably clippy fallout from the previous commit, but I'm NOT going to nit-pick on this sort of thing here.